### ISSUE TITLE
Build issue missing reference

### ISSUE THREAD

**Issue by Liryna** [viciousviper/DokanCloudFS#1]
Hi @viciousviper ,

I finally know on what you was working :smiley: ! This seems to be amazing and a very nice job !

After building CloudFS without issue, I have some error with the build of DokanCloudFS:

```
1>------ Rebuild All started: Project: DokanCloudFS, Configuration: Debug Any CPU ------
1>C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Composition.AttributedModel". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Composition.Hosting". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Composition.Runtime". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Composition.TypedParts". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\CloudDriveFactory.cs(27,14,27,25): error CS0234: The type or namespace name 'Composition' does not exist in the namespace 'System' (are you missing an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\CompositionInitializer.cs(27,14,27,25): error CS0234: The type or namespace name 'Composition' does not exist in the namespace 'System' (are you missing an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\CompositionInitializer.cs(28,14,28,25): error CS0234: The type or namespace name 'Composition' does not exist in the namespace 'System' (are you missing an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(27,14,27,25): error CS0234: The type or namespace name 'Composition' does not exist in the namespace 'System' (are you missing an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\CompositionInitializer.cs(41,25,41,47): error CS0246: The type or namespace name 'ContainerConfiguration' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\CompositionInitializer.cs(43,24,43,39): error CS0246: The type or namespace name 'CompositionHost' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\CompositionInitializer.cs(37,24,37,39): error CS0246: The type or namespace name 'CompositionHost' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(32,6,32,12): error CS0246: The type or namespace name 'Export' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(35,46,35,101): error CS0246: The type or namespace name 'ExportFactory<IAsyncCloudGateway, CloudGatewayMetadata>' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(37,46,37,96): error CS0246: The type or namespace name 'ExportFactory<ICloudGateway, CloudGatewayMetadata>' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(39,10,39,30): error CS0246: The type or namespace name 'ImportingConstructor' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(40,56,40,111): error CS0246: The type or namespace name 'ExportFactory<IAsyncCloudGateway, CloudGatewayMetadata>' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(40,153,40,203): error CS0246: The type or namespace name 'ExportFactory<ICloudGateway, CloudGatewayMetadata>' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(40,32,40,42): error CS0246: The type or namespace name 'ImportMany' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\GatewayManager.cs(40,129,40,139): error CS0246: The type or namespace name 'ImportMany' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\CloudDriveFactory.cs(38,10,38,16): error CS0246: The type or namespace name 'Import' could not be found (are you missing a using directive or an assembly reference?)
2>------ Rebuild All started: Project: DokanCloudFS.Tests, Configuration: Debug Any CPU ------
3>------ Rebuild All started: Project: DokanCloudFS.Mounter, Configuration: Debug Any CPU ------
3>CSC : error CS0006: Metadata file 'C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\bin\Debug\IgorSoft.DokanCloudFS.dll' could not be found
2>C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.CSharp.CurrentVersion.targets(133,9): warning MSB3884: Could not find rule set file "..\..\CloudFileTools\include\TestAssemblies.ruleset".
2>CSC : error CS0006: Metadata file 'C:\Users\liryna\Desktop\DokanCloudFS\DokanCloudFS\bin\Debug\IgorSoft.DokanCloudFS.dll' could not be found
========== Rebuild All: 0 succeeded, 3 failed, 0 skipped ==========
```

It seems that the reference of System.Composition is wrong.

Otherwise I would be very glad to add your project into the dokany list ofc :) !
DokanOperations seems to be well implemented and well made as first seen.

**Comment by viciousviper** [comment]
Nuget references for MEF2 assemblies (System.Composition.*) were missing.
Should be fixed with fe3ea0eec44f035d0917ced9c48e53d35d55849e - please confirm.

**Comment by Liryna** [comment]
:+1: work  thank you !
I am testing onedrive, how to configure the authentication ? (encryptionKey)
